mlflow/data/numpy_dataset.py

```diff
@@ -140,7 +140,12 @@ def from_numpy(
     """
     from mlflow.data.dataset_source_registry import resolve_dataset_source
 
-    resolved_source = resolve_dataset_source(source)
+    if isinstance(source, DatasetSource):
+        resolved_source = source
+    else:
+        resolved_source = resolve_dataset_source(
+            source,
+        )
```

Comment:
This always existed for pandas_dataset. Need to add to numpy_dataset to support passing in `CodeDatasetSource`